pkg/agent/proxy/proxier.go

```diff
@@ -424,6 +423,11 @@ func (p *proxier) installServices() {
 			needUpdateEndpoints = pSvcInfo.SessionAffinityType() != svcInfo.SessionAffinityType() ||
 				pSvcInfo.ExternalPolicyLocal() != svcInfo.ExternalPolicyLocal() ||
 				pSvcInfo.InternalPolicyLocal() != svcInfo.InternalPolicyLocal()
+			// For an existing Service, if both expected Endpoints number and installed Endpoints number are 0, we don't
+			// need to take care of this Service.
+			if len(endpointsToInstall) == 0 && len(endpointsInstalled) == 0 {
+				continue
+			}
```

Comment:
I think it's wrong to add this check. The following code already skips updating Endpoints when there is no change, and it may update Service flows.
If svc port is updated, shouldn't it update the Service flows even if the Endpoints is still empty?

Reply:
You are right. I added some unit tests to cover the case.